extensions/shared/blocks/cover/components.js

```diff
@@ -0,0 +1,66 @@
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { createContext } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import UpgradeNudge from "../../components/upgrade-nudge";
+
+/**
+ * Nudge shows when the user tries to upload a video file.
+ * Unlike the core/video block, handled/extended by the videopress block,
+ * the nudge is not shown permanently.
+ * It's handled by the MediaPlaceholder component
+ * when the user tries to upload a video file.
+ * For this reason, we can't wrap the edit setting
+ * with the wrapPaidBlock() HOC, as the videopress does.
+ *
+ * @param {object}  props - Information about the user.
+ * @param {string}  props.name - Show the Nudge component.
+ * @param {boolean} props.show - Show the Nudge component.
+ * @returns {*} Nudge component or Null.
+ */
+export const JetpackCoverUpgradeNudge = ( { name, show, align } ) =>
+	show
+		? <div className="jetpack-upgrade-nudge-wrapper wp-block" data-align={ align }>
+			<UpgradeNudge
+				plan="value_bundle"
+				blockName={ name }
+				title={ {
+					knownPlan: __( 'To use a video in this block, upgrade to %(planName)s.', 'jetpack' ),
+					unknownPlan: __( 'To use a video in this block, upgrade to a paid plan.', 'jetpack' ),
+				} }
+				subtitle={ false }
+			/>
+		</div>
+		: null;
+
+/**
+ * Cover Media context
+ * Used to connect the CoverEdit with
+ * the Media Replace Flow.
+ */
+export const CoverMediaContext = createContext();
+
+/**
+ * Cover Media Provider will populate the properties
+ * from the CoverEdit to the Media Replace Flow component.
+ *
+ * @param {object}  props - Provider properties.
+ * @param {string}  props.onFilesUpload - Callback function before to upload files.
+ * @param {boolean} props.children - Provider Children.
+ * @returns {*} Provider component.
+ */
+export const CoverMediaProvider = ( { onFilesUpload, children } ) => {
+	return (
+		<CoverMediaContext.Provider value={ {
+			onFilesUpload,
+		} }>
```

Comment:
This creates a new object on each render of the `Provider` because the object `{
			onFilesUpload
		}` will always be a new object instance.

A a result, all components which actively subscribe the `CoverMediaContext` will be re-rendered each time the Provider is rendered. 
If you're just passing the single `onFilesUpload` callback, then consider passing it directly without the wrapping object:

```
<CoverMediaContext.Provider value={ onFilesUpload }>
```
The other issue is that presumably `onFilesUpload` is also a new function reference on each render? This means that the children subscribing to Context will still re-render because the function reference changes on each render.

To solve this, you might like to consider making sure that wherever you _define_ `onFilesUpload` you wrap it in `useCallback` to ensure you always have a reference to the same function object.

Reply:
See also https://reactjs.org/docs/context.html#caveats